From 0224ac01f620ff226bdf877e2b278d20c0fa1b56 Mon Sep 17 00:00:00 2001
From: chenziyang <chenziyang4@huawei.com>
Date: Mon, 3 Jul 2023 14:27:47 +0800
Subject: [PATCH] Fix run-ptrace-mapper test case failed by allowing
 unw_get_proc_info() return unsuccessful result. Because it is not an error
 unw_get_proc_info() does not return the auxiliary information, so we edit
 test case and don't treat it as error.

---
 tests/test-ptrace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test-ptrace.c b/tests/test-ptrace.c
index 846bcd8..32f42e8 100644
--- a/tests/test-ptrace.c
+++ b/tests/test-ptrace.c
@@ -111,7 +111,7 @@ do_backtrace (void)
 	  printf ("%016lx %-32s (sp=%016lx)\n", (long) ip, buf, (long) sp);
 	}
 
-      if ((ret = unw_get_proc_info (&c, &pi)) < 0)
+      if ((ret = unw_get_proc_info (&c, &pi)) < 0 && ret != -UNW_ENOINFO) /* It's possible unw_get_proc_info don't return infomation */
 	panic ("unw_get_proc_info(ip=0x%lx) failed: ret=%d\n", (long) ip, ret);
       else if (verbose)
 	printf ("\tproc=%016lx-%016lx\n\thandler=%lx lsda=%lx",
-- 
2.33.0

